GoogleSignIn/Tests/Unit/GIDAuthorizationResponseHelperTest.m

```diff
@@ -92,78 +101,89 @@ - (void)testGenerateAuthFlowFromAuthorizationResponse_noCode {
+  NSError *expectedError = [NSError errorWithDomain:kGIDVerifyErrorDomain
+                                               code:GIDVerifyErrorCodeUnknown
+                                           userInfo:errorDict];
+
   GIDAuthorizationResponseHandler *responseHandler =
       [[GIDAuthorizationResponseHandler alloc] initWithAuthorizationResponse:nil
                                                                   emmSupport:nil
-                                                                    flowName:GIDFlowNameVerify
+                                                                    flowName:GIDFlowNameVerifyAccountDetail
                                                                configuration:nil
                                                                        error:error];
 
   GIDAuthFlow *authFlow = [responseHandler generateAuthFlowFromAuthorizationResponse];
   XCTAssertNotNil(authFlow);
   XCTAssertNil(authFlow.authState);
   XCTAssertNotNil(authFlow.error);
-  XCTAssertEqual(authFlow.error.code, error.code);
-#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+  XCTAssertEqual(authFlow.error.domain, expectedError.domain);
+  XCTAssertEqual(authFlow.error.code, expectedError.code);
+  XCTAssertEqualObjects(authFlow.error.userInfo, expectedError.userInfo);
 }
+#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
 
-- (void)testMaybeFetchTokenWithAuthFlowError {
+- (void)testMaybeFetchToken_authFlowError {
   NSError *error = [NSError errorWithDomain:kGIDSignInErrorDomain
                                        code:kGIDSignInErrorCodeUnknown
                                    userInfo:nil];
+
+  OIDTokenResponse *tokenResponse = 
+      [OIDTokenResponse testInstanceWithAccessTokenExpiresIn:@(kExpiringAccessToken)];
+  OIDAuthState *authState = [OIDAuthState testInstanceWithTokenResponse:tokenResponse];
+
+  GIDAuthorizationResponseHandlingFake *responseHandler = 
+      [[GIDAuthorizationResponseHandlingFake alloc] initWithAuthState:authState error:nil];
   GIDAuthFlow *authFlow = [[GIDAuthFlow alloc] initWithAuthState:nil
                                                            error:error
                                                       emmSupport:nil
                                                      profileData:nil];
-  GIDAuthorizationResponseHandler *responseHandler = 
-      [[GIDAuthorizationResponseHandler alloc] initWithAuthorizationResponse:nil
-                                                                  emmSupport:kEMMVersion
-                                                                    flowName:GIDFlowNameSignIn
-                                                               configuration:nil
-                                                                       error:nil];
 
   [responseHandler maybeFetchToken:authFlow];
   XCTAssertNil(authFlow.authState);
   XCTAssertNotNil(authFlow.error);
-  XCTAssertNil(authFlow.emmSupport);
+  XCTAssertEqual(authFlow.error, error);
```

Comment:
I think you want to use `XCTAssertEqualObjects`. [XCTAssertEqual](https://developer.apple.com/documentation/xctest/xctassertequal) is for C scalars.